src/discussions/posts/post-editor/PostEditor.jsx

```diff
@@ -1,134 +1,313 @@
+            />
+            <DiscussionPostType
+              value="question"
+              selected={values.postType === 'question'}
+              type={intl.formatMessage(messages.questionType)}
+              icon={<Help />}
+              description={intl.formatMessage(messages.questionDescription)}
+            />
+          </Form.RadioSet>
+          <Form.Group className="py-2 w-50">
+            <Form.Control
+              name="topic"
+              as="select"
+              value={values.topic}
+              onChange={handleChange}
+              onBlur={handleBlur}
+              aria-describedby="topicAreaInput"
+              floatingLabel={intl.formatMessage(messages.topicArea)}
+            >
+              {nonCoursewareTopics.map(topic => (
+                <option key={topic.id} value={topic.id}>{topic.name}</option>
+              ))}
+              {coursewareTopics.map(topic => (
+                <optgroup label={topic.name} key={topic.name}>
+                  {topic.children.map(subtopic => (
+                    <option
+                      key={subtopic.id}
+                      value={subtopic.id}
+                    >{subtopic.name}
+                    </option>
+                  ))}
+                </optgroup>
+              ))}
+            </Form.Control>
+          </Form.Group>
+          <div className="border-bottom my-4" />
+          <Form.Group
+            className="py-2"
+            isInvalid={isFormikFieldInvalid('title', {
+              errors,
+              touched,
+            })}
+          >
+            <Form.Control
+              name="title"
+              type="text"
+              onChange={handleChange}
+              onBlur={handleBlur}
+              aria-describedby="titleInput"
+              floatingLabel={intl.formatMessage(messages.postTitle)}
+              value={values.title}
+            />
+            {isFormikFieldInvalid('title', {
+              errors,
+              touched,
+            })
+            && (
+              <Form.Control.Feedback type="invalid" hasIcon={false}>
+                {intl.formatMessage(messages.titleError)}
+              </Form.Control.Feedback>
+            )}
+          </Form.Group>
+          <Form.Group
+            className="py-2"
+            isInvalid={isFormikFieldInvalid('comment', {
+              errors,
+              touched,
+            })}
+          >
+            <TinyMCEEditor
+              value={values.comment}
+              onEditorChange={formikCompatibleHandler(handleChange, 'comment')}
+              onBlur={formikCompatibleHandler(handleBlur, 'comment')}
+            />
+            {isFormikFieldInvalid('comment', {
+              errors,
+              touched,
+            })
+            && (
+              <Form.Control.Feedback type="invalid" hasIcon={false}>
+                {intl.formatMessage(messages.commentError)}
+              </Form.Control.Feedback>
+            )}
```

Comment:
We can make it a function that will reduce the code. 
{isFormikFieldInvalid('comment', {
              errors,
              touched,
            })
            && (
             displayErrorMessage('commentError')
            )}

const displayErrorMessage = (messageId) => (
 <Form.Control.Feedback type="invalid" hasIcon={false}>
    {intl.formatMessage(messages.[messageId])}
  </Form.Control.Feedback>
)

Reply:
Done